analyzer/src/main/kotlin/PackageManager.kt

```diff
@@ -116,6 +129,14 @@ abstract class PackageManager(
                             return FileVisitResult.SKIP_SUBTREE
                         }
 
+                        if (excludes.isPathExcluded(rootPath.relativize(dir).pathString)) {
+                            logger.info {
```

Comment:
I'm trying to understand why this is implemented here and not in the caller (`Analyzer`).
Am I right this is for optimizing performance (saving accessing some file system nodes for listing file) ?

I'm asking because this solution works only for excluded directory. Excluded definition files residing in non excluded directories won't get filtered out. Having this in the `Analyzer`
like `pkgManager.findManagedFile(..).filter { definitionFile -> pathExcludes.none  { it.matches(definitionFile) }` would work in any case.

Reply:
To be honest, it did not occur to me that the filtering could be done in the `Analyzer`.

You are right, it is probably a bit more efficient to skip searching directory subtrees that are matched by an exclude. I have extended the code in `PackageManager` to handle excluded files as well.

Reply:
Doing so in the `Analyzer` IMO is way simpler than doing so in the `PackageManager`, also because in the `Analyzer` the `repositoryConfigration` containing all needed data is available. So, if the performance thing was not of interrest anyway, would it make sense to move the handling into the `Analyzer` ? (I fear I may be missing something)

Reply:
Well, doing the filtering in the `PackageManager` is not that complicated either. BTW, I never said that performance was not of interest. A main use case behind this new feature is to reduce analysis time and resource usage by preventing unnecessary work.

What may be a bit related, I was wondering why the logic to find definition files is distributed between the `PackageManager` and the `Analyzer` as it currently is. IHMO, it is not in the responsibility of `PackageManager` to implement such a file search algorithm. Should the logic rather be moved completely to `Analyzer`, or maybe even better to a separate class? (But not in the scope of this PR, which is already quite huge.)